src/test/groovy/com/google/protobuf/gradle/ProtobufModelPluginTest.groovy

```diff
@@ -0,0 +1,353 @@
+import org.junit.Rule
+import org.junit.rules.TemporaryFolder
+import spock.lang.Specification
+
+class ProtobufModelPluginTest extends Specification {
+  @Rule
+  final TemporaryFolder testProjectDir = new TemporaryFolder()
+  File buildFile
+
+  def setup() {
+    buildFile = testProjectDir.newFile('build.gradle')
+
+    def pluginClasspathResource = getClass().classLoader.findResource("plugin-classpath.txt")
+    if (pluginClasspathResource == null) {
+      throw new IllegalStateException("Did not find plugin classpath resource, run `testClasses` build task.")
+    }
+
+    def pluginClasspath = pluginClasspathResource.readLines()
+        .collect { it.replace('\\', '\\\\') } // escape backslashes in Windows paths
+        .collect { "'$it'" }
+        .join(", ")
+
+    // Add the logic under test to the test build
+    buildFile << """
+          buildscript {
+            dependencies {
+              classpath files($pluginClasspath)
+            }
+          }
+        """
+
+    // Generate some useful path
+    def userDir = System.getProperty("user.dir").replace("\\", "\\\\")
+    def vendorDir = "$userDir/vendor"
+    def vendorBinaryDir = "$vendorDir/build/binaries"
+
+    // Check to see if the vendor binaries were generated prior to running the test
+    assert new File("$vendorBinaryDir/protocExecutable").exists()
+    assert new File("$vendorBinaryDir/protobufStaticLibrary").exists()
+    assert new File("$vendorBinaryDir/protobufLiteStaticLibrary").exists()
+
+    buildFile << """
+          apply plugin: 'cpp'
+
+          // Repositories doesn't seems to be working for this.
+          binaries.all {
+            if (toolChain instanceof VisualCpp) {
+              cppCompiler.args "/I$vendorDir/protobuf/src"
+              linker.args "/LIBPATH:$vendorBinaryDir/protobufStaticLibrary", "protobuf.lib"
+              linker.args "/LIBPATH:$vendorBinaryDir/protobufLiteStaticLibrary", "protobufLite.lib"
+            } else {
+              cppCompiler.args "-I$vendorDir/protobuf/src"
+              linker.args "-L$vendorBinaryDir/protobufStaticLibrary", "-lprotobuf"
+              linker.args "-L$vendorBinaryDir/protobufLiteStaticLibrary", "-lprotobufLite"
+            }
+          }
+
+          import com.google.protobuf.gradle.*
+          apply plugin: 'com.google.protobuf'
+
+          model {
+            protobuf {
+              protoc {
+                path = "$vendorBinaryDir/protocExecutable/protoc"
+              }
+            }
+          }
+        """
+
+    testProjectDir.newFolder('src', 'main', 'cpp')
+    testProjectDir.newFolder("src", "main", "proto")
+  }
+
+  def "a default protobuf source set exists for configuration"() {
+    given: "a native executable with proto source set configuration"
+    buildFile << """
+          model {
+            components {
+              main(NativeExecutableSpec) {
+                sources {
+                  proto {
+                    source.include "*.proto"
```

Comment:
Yes I am running the wrapper.

If I run it under the root project directory, it won't compile protoc. If I test it under `testProjectNative`, it will try to build protoc, but still end up with linkage errors:
```
linking protoc failed.
/home/zhangkun/github/protobuf-gradle-plugin-shad0w1nk/vendor/build/binaries/protocLibStaticLibrary/libprotocLib.a(command_line_interface.o): In function `google::protobuf::compiler::CommandLineInterface::MemoryOutputStream::MemoryOutputStream(google::protobuf::compiler::CommandLineInterface::GeneratorContextImpl*, std::string const&, bool)':
command_line_interface.cc:(.text+0x1383): undefined reference to `google::protobuf::io::StringOutputStream::StringOutputStream(std::string*)'
command_line_interface.cc:(.text+0x13f6): undefined reference to `google::protobuf::io::ZeroCopyOutputStream::~ZeroCopyOutputStream()'
/home/zhangkun/github/protobuf-gradle-plugin-shad0w1nk/vendor/build/binaries/protocLibStaticLibrary/libprotocLib.a(command_line_interface.o): In function `google::protobuf::compiler::CommandLineInterface::MemoryOutputStream::MemoryOutputStream(google::protobuf::compiler::CommandLineInterface::GeneratorContextImpl*, std::string const&, std::string const&)':
command_line_interface.cc:(.text+0x14a6): undefined reference to `google::protobuf::io::StringOutputStream::StringOutputStream(std::string*)'
command_line_interface.cc:(.text+0x1519): undefined reference to `google::protobuf::io::ZeroCopyOutputStream::~ZeroCopyOutputStream()'
```